jupyter/README.md

```diff
@@ -0,0 +1,417 @@
+
+
+
+
+
+![alt_text](images/image6.png "image_tooltip")
+
+
+Start the container:
+
+
+```
+    $ docker run -v `pwd`:/homedir -p 8888:8888 gcr.io/quantum-291919/jupyter_qsim
+```
+
+
+The output will be something like:
+
+
+```
+    Unable to find image 'gcr.io/quantum-291919/jupyter_qsim:latest' locally
+    latest: Pulling from quantum-291919/jupyter_qsim
+    3c72a8ed6814: Pull complete 
+    2bd71a698eae: Pull complete 
+    Digest: sha256:d0d0040c6ef9925719459e736631c9fec957b94d377009c591d6285143ebb626
+    ...    
+        To access the notebook, open this file in a browser:
+            file:///root/.local/share/jupyter/runtime/nbserver-1-open.html
+        Or copy and paste one of these URLs:
+            http://79804d33f250:8888/?token=d7a53e728e3dff08c6ed12a15810471486b4c15828a3d70a
+         or http://127.0.0.1:8888/?token=d7a53e728e3dff08c6ed12a15810471486b4c15828a3d70a
+```
+
+
+Copy the last URL (it starts with `127.0.0.1`). Replace `127.0.0.1` with the IP address of the VM 
+
+
+[created above](#build-a-container-optimized-vm): 
+
+
+```
+    http://[EXTERNAL_IP_ADDRESS]:8888/?token=7191178ae9aa4ebe1698b07bb67dea1d289cfd0e0b960373
+```
+
+
+Paste that in your browser. You should now see the Jupyter UI:
+
+
+
+
+
+![alt_text](images/image2.png "image_tooltip")
+
+
+Navigate to  qsim > docs > tutorials. You will see:
+
+
+
+
+
+![alt_text](images/image12.png "image_tooltip")
+
+
+Click on qsimcirq.ipynb. This will load the notebook.
+
+You can skip the setup and go straight to the cell that "Full state-vector simulation":
+
+
+
+![alt_text](images/image1.png "image_tooltip")
+
+
+If you choose to modify the notebook, you can save it on the qsim-1 VM from File -> Save As, and saving to /homedir/mynotebook.ipynb.  This will save in your home directory on your VM.
+
+
+
+
+
+![alt_text](images/image4.png "image_tooltip")
+
+
+
+## Run interactively: "Hello Qubit" {#run-interactively-"hello-qubit"}
+
+To run interactively within the container, you can open a second shell window to the VM as you did [above](#open-a-ssh-window-to-your-vm).
+
+Now, find the container ID with docker ps:
+
+
+```
+    $ docker ps
+```
+
+
+
+![alt_text](images/image11.png "image_tooltip")
+
+
+The CONTAINER ID is a UID something like "79804d33f250". Now you can connect to the container:
+
+
+```
+    $ docker exec -it [CONTAINER_ID] /bin/bash
+```
+
+
+This shell is running inside the container, with Python3, cirq and qsim installed. Start python3:
+
+
+```
+	$ python3
+```
+
+
+You will see the output:
+
+
+```
+    Python 3.6.8 (default, Apr 16 2020, 01:36:27) 
+    [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)] on linux
+    Type "help", "copyright", "credits" or "license" for more information.
+    >>>
+```
+
+
+You are now ready for an interactive session running qsim.
+
+
+### Build the circuit {#build-the-circuit}
+
+To import the libraries and build the circuit, copy and paste following into the VM window.
+
+
+```
```

Comment:
For this and other code blocks (not output), open the block with **```python** to add code formatting.

Reply:
fixed.